From ec9561e8163f46c957a3bff6ab876cc91b27ec57 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Fri, 10 Aug 2018 10:04:19 +0530
Subject: [PATCH 1/3] Don't link to libpythonXY.dll/so/dylib

We are not embedding the python interpreter. We are just building a
shared library.

xref: https://gitlab.kitware.com/cmake/cmake/issues/18100
xref: https://groups.google.com/a/continuum.io/forum/#!topic/anaconda/057P4uNWyCU
---
 Malmo/src/CMakeLists.txt               | 3 ---
 Malmo/src/PythonWrapper/CMakeLists.txt | 8 ++++++--
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/Malmo/src/CMakeLists.txt b/Malmo/src/CMakeLists.txt
index 22596bc..70aa3ba 100755
--- a/Malmo/src/CMakeLists.txt
+++ b/Malmo/src/CMakeLists.txt
@@ -116,9 +116,6 @@ set( MALMO_LINK_LIBRARIES
 if( INCLUDE_ALE )
   set( MALMO_LINK_LIBRARIES ${MALMO_LINK_LIBRARIES} ${ALE_LIBRARIES} )
 endif()
-if( INCLUDE_PYTHON )
-  set( MALMO_LINK_LIBRARIES ${MALMO_LINK_LIBRARIES} ${PYTHON_LIBRARIES} )
-endif()
 if( UNIX AND NOT APPLE )
   set( MALMO_LINK_LIBRARIES ${MALMO_LINK_LIBRARIES} rt )
 endif()
diff --git a/Malmo/src/PythonWrapper/CMakeLists.txt b/Malmo/src/PythonWrapper/CMakeLists.txt
index 9088e67..0c3cf04 100755
--- a/Malmo/src/PythonWrapper/CMakeLists.txt
+++ b/Malmo/src/PythonWrapper/CMakeLists.txt
@@ -27,11 +27,15 @@ set( OTHER_FILES
 
 python_add_module( MalmoPython SHARED ${CPP_SOURCES} )
 include_directories( MalmoPython ${PYTHON_INCLUDE_DIRS} )
-target_link_libraries( MalmoPython Malmo ${PYTHON_LIBRARIES} )
+target_link_libraries( MalmoPython Malmo )
 if( MSVC )
   set_property( TARGET MalmoPython APPEND PROPERTY COMPILE_DEFINITIONS "BOOST_PYTHON_STATIC_LIB=1" ) 
+  target_link_libraries( MalmoPython Malmo ${PYTHON_LIBRARIES} )
 elseif( APPLE )
-  set_property( TARGET MalmoPython PROPERTY SUFFIX ".so" )
+  set_target_properties(MalmoPython
+      PROPERTIES
+      SUFFIX ".so"
+      LINK_FLAGS "-undefined dynamic_lookup")
 endif()
 
 # copy the extra files to the build folder for use with testing
-- 
1.8.3.1

